### PR TITLE
Motion State Flags

### PR DESCRIPTION
**Pull request by ogmage78**
* Fixed issue with MotionStateFlag - setting now by the existence of data like we do with all other flags.   This prevents data / flag mismatch

### CODE REVIEW

File: Source/ACE/Entity/MovementData.cs [hunk 1]
```diff
@@ -250,6 +169,29 @@ public MovementData ConvertToClientAccepted(uint holdKey, CreatureSkill run)
             return md;
         }
 
+        public void SetMovementStateFlag()
+        {
```
Comment (ddevec) [inline Source/ACE/Entity/MovementData.cs]: I'm going to be nit-picky here -- I don't like having to remember/know to call this function before we get the MovmentStateFlag.  Can we make the accessor (get) of MovementStateFlag do this computation?

Comment (ogmage78) [inline Source/ACE/Entity/MovementData.cs]: Flags are only used when we serialize - I can move it to the serialize method and call it there?

Comment (ogmage78) [inline Source/ACE/Entity/MovementData.cs]: I made the changes you requested.

Comment (ogmage78) [inline Source/ACE/Entity/MovementData.cs]: It was working the way I had it, but it is not working in accessor.   I am trying to debug now.

Comment (ogmage78) [inline Source/ACE/Entity/MovementData.cs]: I have tried to do it the way you requested - but it does not seem to work.   It seems that it should I must have something obviously wrong.

Comment (ogmage78) [inline Source/ACE/Entity/MovementData.cs]: I found the bug in this.   You can merge it if you want to.

**Pull request merged**